packages/playground/src/components/Playground.tsx

```diff
@@ -0,0 +1,357 @@
+import isEqualWith from 'lodash/isEqualWith';
+
+import { samples } from '../samples';
+import DemoFrame from './DemoFrame';
+import ErrorBoundary from './ErrorBoundary';
+import CopyLink from './CopyLink';
+import GeoPosition from './GeoPosition';
+import ThemeSelector from './ThemeSelector';
+import Selector from './Selector';
+import ValidatorSelector from './ValidatorSelector';
+import SubthemeSelector from './SubthemeSelector';
+import RawValidatorTest from './RawValidatorTest';
+import Editor from './Editor';
+
+const log = (type: string) => console.log.bind(console, type);
+const toJson = (val: unknown) => JSON.stringify(val, null, 2);
+
+const liveSettingsSchema = {
+  type: 'object',
+  properties: {
+    liveValidate: { type: 'boolean', title: 'Live validation' },
+    disable: { type: 'boolean', title: 'Disable whole form' },
+    readonly: { type: 'boolean', title: 'Readonly whole form' },
+    omitExtraData: { type: 'boolean', title: 'Omit extra data' },
+    liveOmit: { type: 'boolean', title: 'Live omit' },
+    noValidate: { type: 'boolean', title: 'Disable validation' },
+    focusOnFirstError: { type: 'boolean', title: 'Focus on 1st Error' },
+    showErrorList: {
+      type: 'string',
+      default: 'top',
+      title: 'Show Error List',
+      enum: [false, 'top', 'bottom'],
+    },
+  },
+};
+
+export const Playground: React.FC<{ themes: any; validators: any }> = ({ themes, validators }) => {
+  const [schema, setSchema] = useState<object>(samples.Simple.schema);
+  const [uiSchema, setUiSchema] = useState<object>(samples.Simple.uiSchema);
+  const [formData, setFormData] = useState<object>(samples.Simple.formData);
+  const [extraErrors, setExtraErrors] = useState<unknown>();
+  const [shareURL, setShareURL] = useState<string | null>(null);
+  const [theme, setTheme] = useState<string>('default');
+  const [subtheme, setSubtheme] = useState<string | null>(null);
+  const [stylesheet, setStylesheet] = useState<string | null>(null);
+  const [validator, setValidator] = useState<string>('AJV8');
+  const [showForm, setShowForm] = useState(false);
+  const [liveSettings, setLiveSettings] = useState({
+    showErrorList: 'top',
+    validate: false,
+    disable: false,
+    readonly: false,
+    omitExtraData: false,
+    liveOmit: false,
+  });
+  const [FormComponent, setFormComponent] = useState<any>(withTheme({}));
+  const [ArrayFieldTemplate, setArrayFieldTemplate] = useState<any>();
+  const [ObjectFieldTemplate, setObjectFieldTemplate] = useState<any>();
+
+  const playGroundFormRef = useRef<any>(null);
+
+  useEffect(() => {
+    const hash = document.location.hash.match(/#(.*)/);
+    if (hash && typeof hash[1] === 'string' && hash[1].length > 0) {
+      try {
+        load(JSON.parse(atob(hash[1])));
+      } catch (err) {
+        alert('Unable to load form setup data.');
+      }
+    } else {
+      // initialize theme
+      onThemeSelected(theme, themes[theme]);
```

Comment:
Do you need to define this before you use it? (i.e. move this above the `useEffect()`)?